src/main/resources/Macros/MapMacro/MapMacro.xml

```diff
@@ -249,36 +249,36 @@ A beautiful street in Paris
     <property>
       <code>{{velocity}}
 ## Get API Key (e.g. for Google and Mapbox)
-#set ($apiKeyDoc = $xwiki.getDocument('MapMacro.Code.Configuration'))
-#set ($apiKeyObj = $apiKeyDoc.getObject('MapMacro.Code.ConfigurationClass'))
+#set ($apiKeyDoc = $xwiki.getDocument('Macros.MapMacro.Code.Configuration'))
+#set ($apiKeyObj = $apiKeyDoc.getObject('Macros.MapMacro.Code.ConfigurationClass'))
 #set ($apiKey = $apiKeyObj.getValue('apiKey'))
 #set ($library = $xcontext.macro.params.library)
 #if ($xcontext.action != 'edit')
   ## The map should not be displayed when the mode is edit, so we don't load the JS in that case.
   ## Without this check, the macro cannot be added into a page with WYSIWYG editor
   ## (see http://jira.xwiki.org/browse/MAP-7).
   #if ($library == "googlemaps")
-    #set ($discard = $xwiki.jsx.use('MapMacro.Code.GoogleMap'))
+    #set ($discard = $xwiki.jsx.use('Macros.MapMacro.Code.GoogleMap'))
   #else
-    #set ($discard = $xwiki.ssx.use('MapMacro.Code.LeafletMap'))
-    #set ($discard = $xwiki.jsx.use('MapMacro.Code.LeafletMap'))
+    #set ($discard = $xwiki.ssx.use('Macros.MapMacro.Code.LeafletMap'))
+    #set ($discard = $xwiki.jsx.use('Macros.MapMacro.Code.LeafletMap'))
   #end
 #end
 #set ($mapId = $util.generateRandomString(5))
 #set ($width = $xcontext.macro.params.width)
 #set ($height = $xcontext.macro.params.height)
-#if ($width.indexOf('px') &lt; 0)
+#if ($width.indexOf('px') &lt; 0 &amp;&amp; $width.indexOf('%') &lt; 0)
```

Comment:
Do we use other different units for expressing length or just `px` and `%`? Why not testing here if the $width is a number instead of looking for `px` and `%`?

Reply:
Only `px` and `%` are used indeed. The new code tests if width/height are numbers.